internal/upstreamoidc/upstreamoidc.go

```diff
@@ -151,32 +151,39 @@ func (p *ProviderConfig) PerformRefresh(ctx context.Context, refreshToken string
 	return p.Config.TokenSource(httpClientContext, &oauth2.Token{RefreshToken: refreshToken}).Token()
 }
 
-// RevokeRefreshToken will attempt to revoke the given token, if the provider has a revocation endpoint.
-func (p *ProviderConfig) RevokeRefreshToken(ctx context.Context, refreshToken string) error {
+// RevokeToken will attempt to revoke the given token, if the provider has a revocation endpoint.
+// It may return an error wrapped by a RetryableRevocationError, which is an error indicating that it may
+// be worth trying to revoke the same token again later. Any other error returned should be assumed to
+// represent an error such that it is not worth retrying revocation later, even though revocation failed.
+func (p *ProviderConfig) RevokeToken(ctx context.Context, token string, tokenType provider.RevocableTokenType) error {
 	if p.RevocationURL == nil {
-		plog.Trace("RevokeRefreshToken() was called but upstream provider has no available revocation endpoint", "providerName", p.Name)
+		plog.Trace("RevokeToken() was called but upstream provider has no available revocation endpoint",
+			"providerName", p.Name,
+			"tokenType", tokenType,
+		)
 		return nil
 	}
 	// First try using client auth in the request params.
-	tryAnotherClientAuthMethod, err := p.tryRevokeRefreshToken(ctx, refreshToken, false)
+	tryAnotherClientAuthMethod, err := p.tryRevokeToken(ctx, token, tokenType, false)
```

Comment:
Consider caching which auth method is successful.

Reply:
We discussed and decided to open a new issue to consider this after the next release. See https://github.com/vmware-tanzu/pinniped/issues/947.